dockstore-webservice/src/main/java/io/dockstore/webservice/DockstoreWebserviceApplication.java

```diff
@@ -486,6 +495,57 @@ public void lifeCycleStarted(LifeCycle event) {
                 }
             }
         });
+
+        environment.lifecycle().addLifeCycleListener(new LifeCycle.Listener() {
+            @Override
+            public void lifeCycleStarted(LifeCycle event) {
+                Session session = hibernate.getSessionFactory().openSession();
+                ManagedSessionContext.bind(session);
+                final Transaction transaction = session.beginTransaction();
+                try {
+                    TokenDAO tokenDAO = new TokenDAO(hibernate.getSessionFactory());
+                    // exclude both gho tokens that the application generates and ghp tokens that we can insert for testing
+                    // https://github.blog/2021-04-05-behind-githubs-new-authentication-token-formats/
+                    tokenDAO.findAllGitHubTokens().stream().filter(t -> !t.getContent().startsWith("gho_") && !t.getContent().startsWith("ghp_")).forEach(token -> {
+                        try {
+                            // cannot use normal github library
+                            final java.net.http.HttpClient client = java.net.http.HttpClient.newBuilder().build();
+                            HttpRequest request = HttpRequest.newBuilder()
+                                .uri(new URI("https://api.github.com/applications/" + configuration.getGithubClientID() + "/token"))
+                                .header("Accept", "application/vnd.github+json")
+                                .header("Authorization", getBasicAuthenticationHeader(configuration.getGithubClientID(), configuration.getGithubClientSecret()))
+                                .method("PATCH", BodyPublishers.ofString("{\"access_token\":\"" + token.getContent() + "\"}"))
+                                .build();
+                            final HttpResponse<ResetTokenModel> send = client.send(request, new JsonBodyHandler<>(ResetTokenModel.class));
+                            final String token1 = send.body().token;
+                            if (send.statusCode() != HttpStatus.SC_NOT_FOUND) {
+                                token.setContent(token1);
+                                tokenDAO.update(token);
+                                LOG.info("updated token for {}", token.getUsername());
+                            } else {
+                                LOG.error("could not update old style github token for {}, token was not found on github", token.getUsername());
+                            }
+                        } catch (IOException e) {
+                            LOG.error("could not update old style github token for {}", token.getUsername());
```

Comment:
It would be good to log the exception, although it's a pain because you can use LOG's formatting functionality if you do so.